aziot/src/internal/check/checks/cert_expiry.rs

```diff
@@ -0,0 +1,241 @@
+                    let (bootstrap_cert_res, bootstrap_certificate_info) =
+                        validate_cert(certd_config, bootstrap_cert_id, bootstrap_cert_name).await?;
+                    self.bootstrap_certificate_info = bootstrap_certificate_info;
+
+                    if !matches!(bootstrap_cert_res, CheckResult::Ok) {
+                        return Ok(bootstrap_cert_res);
+                    }
+                }
+
+                Ok(CheckResult::Ok)
+            }
+            None => Ok(CheckResult::Ignored),
+        }
+    }
+}
+
+#[derive(Serialize, Default)]
+struct LocalCaCert {
+    certificate_info: Option<CertificateValidity>,
+}
+
+#[async_trait::async_trait]
+impl Checker for LocalCaCert {
+    fn meta(&self) -> CheckerMeta {
+        CheckerMeta {
+            id: "local-ca-certificate-expiry",
+            description: "production readiness: Local CA certificates expiry",
+        }
+    }
+
+    async fn execute(&mut self, shared: &CheckerShared, cache: &mut CheckerCache) -> CheckResult {
+        self.inner_execute(shared, cache)
+            .await
+            .unwrap_or_else(CheckResult::Failed)
+    }
+}
+
+impl LocalCaCert {
+    async fn inner_execute(
+        &mut self,
+        _shared: &CheckerShared,
+        cache: &mut CheckerCache,
+    ) -> Result<CheckResult> {
+        let certd_config = unwrap_or_skip!(&cache.cfg.certd);
+
+        let cert_id = certd_config
+            .cert_issuance
+            .local_ca
+            .as_ref()
+            .map(|local_ca| &local_ca.cert);
+        let cert_id = match cert_id {
+            Some(id) => id,
+            None => return Ok(CheckResult::Ignored),
+        };
+
+        let (res, cert_info) = validate_cert(certd_config, cert_id, "Local CA").await?;
+        self.certificate_info = cert_info;
+        Ok(res)
+    }
+}
+
+/// Validate the certificate is valid, returning `CheckResult::Ok` if the
+/// certificate is configured to be dynamically issued, but hasn't been issued
+/// yet (i.e: doesn't correspond to a file on-disk yet).
+///
+/// `cert_name` is only used for friendly error messages.
+async fn validate_cert(
+    certd_config: &aziot_certd_config::Config,
+    cert_id: &str,
+    cert_name: &str,
+) -> anyhow::Result<(CheckResult, Option<CertificateValidity>)> {
+    let path = aziot_certd_config::util::get_path(
+        &certd_config.homedir_path,
+        &certd_config.preloaded_certs,
+        cert_id,
+        false,
+    )
+    .map_err(|e| anyhow!("{}", e))?;
+
+    if path.exists() {
+        let cert_info = CertificateValidity::new(path, cert_name, cert_id).await?;
+        cert_info
+            .to_check_result()
+            .map(|res| (res, Some(cert_info)))
+    } else if !certd_config.preloaded_certs.contains_key(cert_id)
+        && !certd_config.cert_issuance.certs.contains_key(cert_id)
+    {
+        Err(anyhow!(
+            "{} certificate is not manually specified, nor is it dynamically issued",
```

Comment:
`"{} certificate is neither preloaded nor configured to be dynamically issued, and thus cannot be used."`